mlflow/store/rest_store.py

```diff
@@ -152,6 +152,7 @@ def create_run(self, experiment_id, user_id, run_name, source_type, source_name,
         response_proto = self._call_endpoint(CreateRun, req_body)
         run = Run.from_proto(response_proto.run)
         if run_name:
+            # TODO: optimization: This is making 2 calls to backend store. Include with above call.
```

Comment:
We can add this to `user_specified_tags` and will `ContextProviders` will auto resolve this as run name. This is a bit of a involved change and not related to this fix. Will handle this separately.